config.go

```diff
@@ -2111,3 +2111,79 @@ func checkEstimateMode(estimateMode string) error {
 	return fmt.Errorf("estimatemode must be one of the following: %v",
 		bitcoindEstimateModes[:])
 }
+
+// configToFlatMap converts the given config struct into a flat map of key/value
+// pairs using the dot notation we are used to from the config file or command
+// line flags.
+func configToFlatMap(cfg Config) (map[string]string, error) {
```

Comment:
The only major concern I have with this is if we can do a `Config -> Config` transformation that redacts sensitive info that is irrelevant wrt debugging. The one that comes to mind is any `bitcoind` `rpcpass` or stuff like that.

Reply:
Very good point, those values don't really need to be known by a supporting person. And if a special character in those fields cause the config file not to be parsed correctly, it should be noticeable in another way.

Reply:
I added a string suffix based redaction, which covers all the keys you mentioned (except for `Bitcoind.RPCAuth` which doesn't exist on the `lnd` side, only in `bitcoind`'s config).